library/Vanilla/DateFilterSchema.php

```diff
@@ -314,4 +314,67 @@ public static function dateFilterField($field, array $dateData) {
 
         return $result;
     }
+
+    /**
+     * If the parameter value is a valid date filter value,
+     * return a structure: startDate, endDate, exclude flag.
+     * Useful for Sphinx setFilterRange
+     *
+     * @param mixed $dateData The decoded and validated date data.
+     *
+     * @throws Exception If dateData structure is wrong throw exception.
+     *
+     * @return array Structure: DateTime startDate, DateTime endDate, Boolean inclusive
+     */
+    public static function dateFilterRange(array $dateData) {
+        $validOperators = ['=', '>', '<', '>=', '<=', '[]', '()'];
+        $result = [
+            'startDate' => (new \DateTime())->setDate(1970, 1, 1),
+            'endDate' => (new \DateTime())->setDate(2100, 12, 31), // '2100-12-12'
```

Comment:
Because Sphinx setFilterRange method accepts only integers

http://sphinxsearch.com/docs/current/api-func-setfilterrange.html

so we need to send min and max value anyway.

Reply:
If this reasoning is specific to Sphinx, shouldn't it be implemented in the code where Sphinx is being used? This looks like a general-purpose method.

Reply:
This whole class pretend to be "general-purpose". Adam did ask to use it since it is general.
And this class has `dateFilterFiled` method
https://github.com/vanilla/vanilla/blob/ad52ea99664bea5add6dc004804e38275a4dd45e/library/Vanilla/DateFilterSchema.php#L270?

which looks like general-purpose method, but it is not. It is **SQL WHERE part builder**

IMHO I don't break any OOP logic here.

Reply:
The doc block on `dateFilterField` says:

> If the parameter value is a valid date filter value, return an array of query conditions.

To me, that says this function was built for SQL queries. Conversely, the doc block on this method says:

> If the parameter value is a valid date filter value, return a structure: startDate, endDate, exclude flag. Useful for Sphinx setFilterRange

This does not indicate, to me, that this method has behavior specific for Sphinx. It says the result could be used for Sphinx requests. My suggestion here is to pull out the stuff specific to Sphinx (e.g. the arbitrary default start/end dates) and add those where you actually need them (e.g. when making a Sphinx request). Keeping them in here also means I can't provide my own defaults, wherever I'm invoking this method, because the result will always have a valid start/end date.